libs/server-sdk/src/data_sources/data_source_event_handler.cpp

```diff
@@ -0,0 +1,218 @@
+#include "data_source_event_handler.hpp"
+
+#include <launchdarkly/encoding/base_64.hpp>
+#include <launchdarkly/serialization/json_flag.hpp>
+#include <launchdarkly/serialization/json_primitives.hpp>
+#include <launchdarkly/serialization/json_sdk_data_set.hpp>
+#include <launchdarkly/serialization/json_segment.hpp>
+#include <launchdarkly/serialization/value_mapping.hpp>
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/json.hpp>
+#include <unordered_map>
+
+#include <utility>
+
+#include "tl/expected.hpp"
+
+namespace launchdarkly::server_side::data_sources {
+
+static char const* const kErrorParsingPut = "Could not parse PUT message";
+static char const* const kErrorPutInvalid =
+    "PUT message contained invalid data";
+static char const* const kErrorParsingPatch = "Could not parse PATCH message";
+static char const* const kErrorPatchInvalid =
+    "PATCH message contained invalid data";
+static char const* const kErrorParsingDelete = "Could not parse DELETE message";
+static char const* const kErrorDeleteInvalid =
+    "DELETE message contained invalid data\"";
+
+template <typename TStreamingDataKind, typename TData>
+tl::expected<DataSourceEventHandler::Patch, JsonError> Patch(
+    std::string const& path,
+    boost::json::object const& obj) {
+    auto const* data_iter = obj.find("data");
+    if (data_iter == obj.end()) {
+        return tl::unexpected(JsonError::kSchemaFailure);
+    }
+    auto data =
+        boost::json::value_to<tl::expected<std::optional<TData>, JsonError>>(
+            data_iter->value());
+    if (!data.has_value()) {
+        return tl::unexpected(JsonError::kSchemaFailure);
+    }
+    return DataSourceEventHandler::Patch{
+        TStreamingDataKind::Key(path),
+        data_model::ItemDescriptor<TData>(data->value())};
```

Comment:
I wish this lint worked, because it would be useful, but it is basically always wrong. Checked on line 41.